neo/Network/P2P/Payloads/Block.cs

```diff
@@ -51,9 +55,16 @@ public static UInt256 CalculateMerkleRoot(UInt256 consensusDataHash, params UInt
         public override void Deserialize(BinaryReader reader)
         {
             base.Deserialize(reader);
+            int count = (int)reader.ReadVarInt(MaxContentsPerBlock);
+            if (count == 0) throw new FormatException();
             ConsensusData = reader.ReadSerializable<ConsensusData>();
-            Transactions = reader.ReadSerializableArray<Transaction>(MaxTransactionsPerBlock);
-            if (Transactions.Length == 0) throw new FormatException();
+            Transactions = new Transaction[count - 1];
+            for (int i = 0; i < Transactions.Length; i++)
+            {
+                Transactions[i] = reader.ReadSerializable<Transaction>();
+                if (Transactions[i].Hash.Equals(ConsensusData.Hash))
```

Comment:
This could be possible?

Reply:
It is impossible now. But if the `ConsensusData` is expanded in the future, it will be possible.

Reply:
But this doesn't affect us? what will be the effect of a duplicated hash in something that is not the same thing?

Reply:
Okey, I will remove the check.